generator/internal/sidekick/command.go

```diff
@@ -0,0 +1,256 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sidekick
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"slices"
+	"strings"
+	"text/template"
+)
+
+// A Command is an implementation of a sidekick command
+// like sidekick generate
+type Command struct {
+	action           func(rootConfig *Config, cmdLine *CommandLine) error
+	usageLine        string
+	altNames         []string
+	shortDescription string
+	longDescription  string
+	flags            *flag.FlagSet
+	commands         []*Command
+	parent           *Command
+}
+
+func (c *Command) UsageLine() string {
+	return c.usageLine
+}
+
+func (c *Command) LongDescription() string {
+	if c.longDescription != "" {
+		return c.longDescription
+	} else {
+		return c.shortDescription
+	}
+}
+
+func (c *Command) ShortDescription() string {
+	return c.shortDescription
+}
+
+// LongName returns the command's long name: all the words in the usage line between "sidekick" and a flag or argument,
+func (c *Command) LongName() string {
+	name := c.usageLine
+	if i := strings.Index(name, " ["); i >= 0 {
+		name = name[:i]
+	}
+	return strings.TrimSpace(strings.TrimPrefix(name, "sidekick"))
+}
+
+func (c *Command) AddAltName(n string) *Command {
+	c.altNames = append(c.altNames, n)
+	return c
+}
+
+// Name returns the command's short name: the last word in the usage line before a flag or argument.
+func (c *Command) Name() string {
+	name := c.LongName()
+	if i := strings.LastIndex(name, " "); i >= 0 {
+		name = name[i+1:]
+	}
+	return name
+}
+
+func (c *Command) Names() []string {
+	names := []string{c.Name()}
+	names = append(names, c.altNames...)
+	return names
+}
+
+func (c *Command) Runnable() bool {
```

Comment:
It doesn't look like this is used anywhere

Reply:
removed